src/ride/vehicle_paint.c

```diff
@@ -886,6 +887,1416 @@ const vehicle_boundbox VehicleBoundboxes[16][224] = {
+	vehicle_sprite_paint(vehicle, ebx, ecx, z, vehicleEntry);
+}
+
+void vehicle_sprite_paint_6D51DE(rct_vehicle *vehicle, int ebx, int z, const rct_ride_entry_vehicle *vehicleEntry) //6D51DE
+{
+	if (vehicle->restraints_position < 64) {
+		vehicle_sprite_paint_6D51EB(vehicle, ebx, z, vehicleEntry);
+		return;
+	}
+	if (!(vehicleEntry->sprite_flags & 0x2000)) {
+		vehicle_sprite_paint_6D51EB(vehicle, ebx, z, vehicleEntry);
+		return;
+	}
+	if (ebx & 3) {
+		vehicle_sprite_paint_6D51EB(vehicle, ebx, z, vehicleEntry);
+		return;
+	}
+	int ecx = ebx / 2;
+	ebx = ebx / 8;
+	ebx += ((vehicle->restraints_position - 64) / 64) * 4;
+	ebx *= vehicleEntry->var_16;
+	ebx += vehicleEntry->var_1C;
+	vehicle_sprite_paint(vehicle, ebx, ecx, z, vehicleEntry);
+}
+
+void vehicle_sprite_0_0(rct_vehicle *vehicle, int imageDirection, int z, const rct_ride_entry_vehicle *vehicleEntry) //6D51DE
+{
+	vehicle_sprite_paint_6D51DE(vehicle, imageDirection, z, vehicleEntry);
+}
+
+void vehicle_sprite_0_1(rct_vehicle *vehicle, int imageDirection, int z, const rct_ride_entry_vehicle *vehicleEntry) //6D4EE7
+{
+	if (vehicleEntry->sprite_flags & 0x20) {
+		int ecx = imageDirection / 2;
+		int ebx = ((imageDirection / 4) * vehicleEntry->var_16) + vehicleEntry->var_30;
+		vehicle_sprite_paint_6D520E(vehicle, ebx, ecx, z, vehicleEntry);
+	} else {
+		vehicle_sprite_paint_6D51DE(vehicle, imageDirection, z, vehicleEntry);
+	}
+}
+
+void vehicle_sprite_0_2(rct_vehicle *vehicle, int imageDirection, int z, const rct_ride_entry_vehicle *vehicleEntry) //6D4F34
+{
+	if (vehicleEntry->sprite_flags & 0x20) {
+		int ecx = (imageDirection / 2) + 108;
+		int ebx = ((imageDirection + 16) * vehicleEntry->var_16) + vehicleEntry->var_30;
+		vehicle_sprite_paint_6D520E(vehicle, ebx, ecx, z, vehicleEntry);
+	} else {
+		vehicle_sprite_paint_6D51DE(vehicle, imageDirection, z, vehicleEntry);
+	}
+}
+
+void vehicle_sprite_0_3(rct_vehicle *vehicle, int imageDirection, int z, const rct_ride_entry_vehicle *vehicleEntry) //6D4F0C
+{
+	if (vehicleEntry->sprite_flags & 0x20) {
+		int ecx = imageDirection / 2;
+		int ebx = (((imageDirection / 4) + 8) * vehicleEntry->var_16) + vehicleEntry->var_30;
+		vehicle_sprite_paint_6D520E(vehicle, ebx, ecx, z, vehicleEntry);
+	} else {
+		vehicle_sprite_paint_6D51DE(vehicle, imageDirection, z, vehicleEntry);
+	}
+}
+
+void vehicle_sprite_0_4(rct_vehicle *vehicle, int imageDirection, int z, const rct_ride_entry_vehicle *vehicleEntry) //6D4F5C
+{
+	if (vehicleEntry->sprite_flags & 0x20) {
+		int ecx = ((imageDirection / 2) ^ 8) + 108;
+		int ebx = ((imageDirection + 48) * vehicleEntry->var_16) + vehicleEntry->var_30;
+		vehicle_sprite_paint_6D520E(vehicle, ebx, ecx, z, vehicleEntry);
+	} else {
+		vehicle_sprite_paint_6D51DE(vehicle, imageDirection, z, vehicleEntry);
+	}
+}
+
+void vehicle_sprite_0_5(rct_vehicle *vehicle, int imageDirection, int z, const rct_ride_entry_vehicle *vehicleEntry) //6D4F84
+{
+	if (vehicle->update_flags & 0x800) {
+		vehicleEntry--;
```

Comment:
This might cause issues as the `vehicle_sprite_paint()` function previously reloaded the old vehicle entry. I'm unsure if anything would actually be effected by this though.